compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/GPULoweringConfigUtils.cpp

```diff
@@ -127,37 +127,34 @@ getPromotedOperandList(LoweringConfigAttr config) {
   return getIntegerVector(array);
 }
 
-std::optional<SmallVector<bool>> getUseDirectLoad(LoweringConfigAttr config) {
-  auto array = config.getAttributes().getAs<ArrayAttr>(kDirectLoadOperandsName);
+std::optional<ArrayRef<Attribute>>
+getPromotionTypesList(LoweringConfigAttr config) {
+  auto array = config.getAttributes().getAs<ArrayAttr>(kPromotionTypesName);
   if (!array) {
     return std::nullopt;
   }
-  if (!llvm::all_of(array.getValue(), llvm::IsaPred<BoolAttr>)) {
-    return std::nullopt;
-  }
-  return llvm::map_to_vector(array.getValue(), [](Attribute s) -> bool {
-    return cast<BoolAttr>(s).getValue();
-  });
+  return array.getValue();
 }
 
 void appendPromotedOperandsList(MLIRContext *context,
                                 SmallVectorImpl<NamedAttribute> &attrs,
                                 ArrayRef<int64_t> operands,
-                                ArrayRef<bool> directLoadOperands) {
+                                ArrayRef<Attribute> promotionTypes) {
   Builder b(context);
   attrs.emplace_back(StringAttr::get(context, kPromoteOperandsName),
                      b.getI64ArrayAttr(operands));
-  if (!directLoadOperands.empty()) {
-    assert(directLoadOperands.size() == operands.size() &&
-           "Direct load operands size must match promoted operands size");
-    attrs.emplace_back(StringAttr::get(context, kDirectLoadOperandsName),
-                       b.getBoolArrayAttr(directLoadOperands));
+  if (!promotionTypes.empty()) {
+    assert(promotionTypes.size() == operands.size() &&
+           "Promotion types size must match promoted operands size");
+    attrs.emplace_back(StringAttr::get(context, kPromotionTypesName),
+                       b.getArrayAttr(promotionTypes));
```

Comment:
You don't need to construct the StringAttr passed to NamedAttr, it's enough to give it a StringRef